interfaces/builtin/README.md

```diff
@@ -0,0 +1,305 @@
+# Interfaces static information
+
+## BaseDeclaration{Plugs,Slots}
+
+The snap-declarations are mainly used to control what plugs or slots a snap is
+allowed to use, and if a snap is allowed to use a plug/slot, what other
+slots/plugs should connect to that plug/slot on this snap.
+
+The behaviour is defined by a map which has 6 possible keys:
+
+- allow-installation
+- deny-installation
+- allow-connection
+- deny-connection
+- allow-auto-connection
+- deny-auto-connection
+
+Each of these keys can either have a static value of true/false or can be a
+more complex object/list which then is “evaluated” by snapd on a device to
+determine the actual value, be it true or false.
+
+Note that the declarations defined in the snapd interfaces can be overridden by
+“store assertions”, which make it possible for the snap store to alter the
+behaviour hardcoded in snapd; for example, a store assertion is typically used
+to grant auto-connection to some plugs of a specific application where this is
+deemed reasonable or safe (for example, auto-connecting the `camera` interface
+in a web-streaming application).
+
+The default value of the `allow-*` keys when not otherwise specified is `true`
+and the default value of `deny-*` keys when not otherwise specified is `false`.
+Matching `deny-*` constraints override/take precedence over `allow-*`
+constraints.
```

Comment:
Is this actually true? That would mean that a `deny-` rule hardcoded in a snapd interface cannot be overruled by the snap store -- which I find unlikely.